app/models/grda_warehouse/auth_policies/policy_provider.rb

```diff
@@ -0,0 +1,108 @@
+###
+# Copyright 2016 - 2024 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+require 'memery'
+
+class GrdaWarehouse::AuthPolicies::PolicyProvider
+  include Memery
+  attr_reader :user
+
+  # TODO: START_ACL remove after ACL migration is complete
+  ###
+  # The legacy permission policy only checks the permissions on a user's roles and assumes the loaded record is visible
+  # to the user already. This is a dangerous assumption so it's required to confirm the record is authorized by
+  # setting legacy_implicitly_assume_authorized_access to true.
+  # Note this is only relevant for users where using_acls is false
+  ###
+  attr_accessor :legacy_implicitly_assume_authorized_access
+  # END_ACL
+
+  def initialize(user)
+    @user = user
+  end
+
+  def for_client(client_or_id)
+    client_id = id_from_arg(client_or_id, GrdaWarehouse::Hud::Client)
+    if user.using_acls?
+      client_project_policy(client_id)
+    else
+      # TODO: START_ACL remove after ACL migration is complete
+      legacy_user_role_policy
+      # END_ACL
+    end
+  end
+
+  def for_patient(patient)
+    for_client(patient.client_id)
+  end
+
+  memoize def for_project(project_or_id)
+    project_id = id_from_arg(project_or_id, GrdaWarehouse::Hud::Project)
+    if user.using_acls?
+      if project_id.in?(viewable_project_ids)
+        GrdaWarehouse::AuthPolicies::ProjectPolicy.new(user: user, project_id: project_id)
+      else
+        deny_policy
+      end
+    else
+      # TODO: START_ACL remove after ACL migration is complete
+      legacy_user_role_policy
+      # END_ACL
+    end
+  end
+
+  protected
+
+  def handle_legacy_unauthorized
+    raise 'legacy authorization not performed'
+  end
+
+  memoize def viewable_project_ids
+    user.viewable_project_ids(:can_view_projects)
```

Comment:
This is the issue I was hinting at.  The permission that you need to check is the permission in question (maybe `can_view_client_names`)  Can view projects is only to be used in the context of if you can see the project.

Reply:
That's compounded by the concept of ROI which can release a client regardless of project relationship.  BUT we don't have a definition of _what_ is being released, so maybe we shouldn't release name or SSN or HIV status....